### PR TITLE
Tests for blocking path pastes and intranet URLs for secondary files...

### PR DESCRIPTION
**Pull request by jmchilton**
... during upload. Clears up a worry I had about some stuff added this release and ensures there no regressions in the security of this API. 

xref #4289 (blocking intranet URLs by default)
xref #4404 (implementing path pastes for normal uploads with the file:// prefix).
xref #4417 (initial tests for blocking file:// prefixes for non-admins added in #4404)

### CODE REVIEW

File: test/integration/test_upload_configuration_options.py [hunk 1]
```diff
@@ -64,6 +67,24 @@ def test(self):
...
+    @skip_without_datatype("velvet")
+    def test_composite_datatype(self):
+        payload = self.dataset_populator.upload_payload(
+            self.history_id,
+            "sequences content",
+            file_type="velvet",
+            extra_inputs={
+                "files_1|url_paste": "roadmaps content",
+                "files_1|type": "upload_dataset",
+                "files_2|url_paste": "file://%s/1.txt",
```
Comment (nsoranzo) [inline test/integration/test_upload_configuration_options.py]: The `%s` doesn't seem right here.

Comment (jmchilton) [reply test/integration/test_upload_configuration_options.py]: Yeah - good point. One second...

Comment (jmchilton) [reply test/integration/test_upload_configuration_options.py]: Rebased with a fix for that - it doesn't change the test outcome really but it definitely was support to be an actual valid file URI. Thanks for the catch.

File: test/integration/test_upload_configuration_options.py [hunk 2]
```diff
@@ -64,6 +67,24 @@ def test(self):
         # the newer API decorator that handles those details.
         assert create_response.status_code >= 400
 
+    @skip_without_datatype("velvet")
+    def test_composite_datatype(self):
```
Comment (nsoranzo) [inline test/integration/test_upload_configuration_options.py]: Should this method be called `test_secondary_file` (or alternatively `test_blocked_url_secondary_file` renamed to `test_blocked_url_composite_datatype`)?

**Comment by nsoranzo** [comment]
Thanks @jmchilton!

**Pull request merged**